packages/rest/src/http/route.ts

```diff
@@ -248,36 +249,27 @@ export function getOperationParameters(
       result.parameters.push({ type: "path", name: pathParam, param });
     } else if (headerParam) {
       result.parameters.push({ type: "header", name: headerParam, param });
-    } else if (bodyParm) {
-      if (result.body === undefined) {
-        result.body = param;
+    } else if (bodyParam) {
+      if (result.bodyType === undefined) {
+        result.bodyParameter = param;
+        result.bodyType = param.type;
       } else {
         diagnostics.add(createDiagnostic({ code: "duplicate-body", target: param }));
       }
     } else {
-      if (unAnnotatedParam === undefined) {
-        unAnnotatedParam = param;
-      } else {
-        diagnostics.add(
-          createDiagnostic({
-            code: "duplicate-body",
-            messageId: "duplicateUnannotated",
-            target: param,
-          })
-        );
-      }
+      unannotatedParam = true;
     }
   }
 
-  if (unAnnotatedParam !== undefined) {
-    if (result.body === undefined) {
-      result.body = unAnnotatedParam;
+  if (unannotatedParam) {
+    if (result.bodyType === undefined) {
+      result.bodyType = operation.parameters;
```

Comment:
shouldn't only the unanotted params be included here?

Reply:
This works because we ignore the metadata elsewhere. I think a comment would be sufficient, I'll add one.

Reply:
On second thought, I will filter them because the API would be confusing otherwise. It also happened to make resurrecting that test easier :)